### PR TITLE
Add speakers list page, improve sessions/tutorials/posters/talks pages

### PR DESCRIPTION
**Pull request by egeakman**
**Preview:** https://ep2024-git-fix-and-new-stuff-europython-society.vercel.app/

- Improved filtering feature:
  - added level field,
  - added a reset button
  - extracted reused parts to components

- Added the speakers list page similar to last year's page

- Added checks to be able to hide empty sections (some sessions don't have speakers, like announcements)

### CODE REVIEW

**Comment by egeakman** [comment]
We probably can make new components to avoid having the same/similar code among the pages

**Comment by artcz** [comment]
Hey, just one quick comment – the filters on the posters and tutorials page were removed on purpose. There is just not enough content there to filter. For talks I guess we could re-add them, although I was also thinking that we should probably have something different there to filter than dropdowns from /sessions – perhaps group them by track or something like that.

Thanks for adding the speakers page!

**Comment by egeakman** [comment]
@artcz Hey! I've changed the structure of the talks page like you said (if I understood what you meant correctly)

File: src/pages/talks.astro [hunk 1]
```diff
@@ -29,57 +30,156 @@ const description = "A list of all the sessions at the conference";
...
+    resetButton?.addEventListener("click", () => {
+      form.querySelectorAll("select").forEach((select) => {
+        select.value = "";
+      });
+      filterSessions();
+    });
+  });
+
+  filterSessions();
+</script>
```
[the remark this answers is not in the document]
Comment (egeakman) [reply src/pages/talks.astro:185]: We probably can, I'll look into it

[the remark this answers is not in the document]
Comment (egeakman) [reply src/pages/talks.astro]: Done!

**Comment by egeakman** [comment]
Split into smaller PRs: #639 #640 #641 #642

**Pull request closed without merging**